### DISCUSSION TITLE
How to use Mac M1 GPU

### DISCUSSION THREAD

**Discussion by runski** [microsoft/onnxruntime#23056] [General] [answered]
I read that onnx runtime v1.19.4 supports acceleration for Apple Silicon.  I assume this means GPU support is now available for the M1 chip.  I can't find any examples of how to specify an M1 gpu in onnx runtime.

If I specify `print(ort.get_device())`, it will show me a CPU device.

If I try to install onnxruntime-gpu, it can't find a version.  I think this is the wrong approach, as onnxruntme-gpu refers to CUDA only.

**Comment by robertknight** [comment, marked as answer]
The way to use the GPU on Apple systems is via the [CoreML execution provider](https://onnxruntime.ai/docs/execution-providers/CoreML-ExecutionProvider.html). This may also use the NPU.
